components/list/list.js

```diff
@@ -0,0 +1,71 @@
+import './list-item.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+
+class List extends LitElement {
+	static get properties() {
+		return {
+			divider: { type: String }
+		};
+	}
+
+	static get styles() {
+		const layout = css`
+			:host {
+				display: block;
+			}
+			ul {
+				list-style-type: none;
+				padding: 0;
+			}
+			::slotted(*) {
+				display: none;
+			}
+			::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: block;
+				--d2l-list-item-divider-top: block;
+				box-sizing: content-box;
+				display: list-item;
+				margin: 1px 0;
+			}
+		`;
+
+		const specialDividers = css`
+			:host([divider="none"]) ::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: none;
+				--d2l-list-item-divider-top: none;
+			}
+			:host([divider="bottom"]) ::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: block;
+				--d2l-list-item-divider-top: none;
+			}
+			:host([divider="top"]) ::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: none;
+				--d2l-list-item-divider-top: block;
+			}
+			:host([divider="middle"]) ::slotted(d2l-list-item:first-of-type),
+			:host([divider="middle"]) ::slotted(d2l-list-item:last-of-type) {
+				--d2l-list-item-divider-bottom: none;
+				--d2l-list-item-divider-top: none;
+			}
+			:host([divider-extend]) ::slotted(d2l-list-item) {
```

Comment:
If `divider-extend` is an attribute they can set, it should be called out above in the property list.

I was kind of hoping we'd be able to automatically extend the dividers based on the variants (e.g. the divider type and whether or not there's selection and/or drag & drop), but if that's super hard to do, I guess we could settle for having an attribute like this. The problem with having an attribute is that it means developers are deciding whether to extend the dividers, instead of it being something we do automatically based on certain conditions.

Reply:
So this is for the "tabular" list type. So it will auto extend for drag and drop and selection.